### PR TITLE
Enforce event age restrictions in registration and admin forms

### PR DESCRIPTION
**Pull request by EuphoriaLux**
## Purpose
Fix a security bypass where users could register for age-restricted events without proper age verification. Previously, the `event_register` view only checked for profile existence but never validated the user's actual age against `event.min_age` and `event.max_age`. This change enforces age restrictions at three levels:

1. **View-level enforcement** (`event_register`): Reject registrations from users whose age falls outside the event's range
2. **Admin form validation** (`EventRegistrationAdminForm`): Prevent admins from manually registering ineligible users
3. **Bulk action protection** (`confirm_registrations`): Skip age-ineligible registrations when bulk-confirming

## Does this introduce a breaking change?
```
[ ] Yes
[x] No
```

## Pull Request Type
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### `crush_lu/views_events.py`
- Enhanced `event_register` to detect age-restricted events (`min_age > 18` or `max_age < 99`)
- For age-restricted events, require a profile with a valid date of birth (not just a checkbox)
- Validate user age against event bounds before registration
- Added defense-in-depth re-verification under transaction lock to prevent race conditions
- Clarified that self-attestation checkbox only applies to open events (no age restriction)

### `crush_lu/admin/events.py`
- Created `EventRegistrationAdminForm` with age validation in `clean()` method
- Validates that users meet event age requirements when admins create/edit registrations
- Allows cancelled registrations to bypass the gate (for historical cleanup)
- Applied form to both `EventRegistrationInline` and `EventRegistrationAdmin`
- Updated `confirm_registrations` bulk action to skip age-ineligible registrations with a warning message

### `crush_lu/tests/test_events.py`
- Added `EventAgeEnforcementOnRegisterTests` class with 8 test cases covering:
  - Rejection of users below/above age range
  - Acceptance of users within range
  - Boundary conditions (exactly at min/max age)
  - Profile-less users blocked from age-restricted events
  - Open events (18-99) don't force profile creation
- Added `EventRegistrationAdminFormAgeTests` class with 4 test cases covering:
  - Admin form rejects underage users
  - Admin form rejects users without DOB
  - Admin form accepts eligible users
  - Cancelled status bypasses age gate

## How to Test
```
python manage.py test crush_lu.tests.test_events.EventAgeEnforcementOnRegisterTests
python manage.py test crush_lu.tests.test_events.EventRegistrationAdminFormAgeTests
```

## What to Check
Verify that:
* Users below `event.min_age` cannot register
* Users above `event.max_age` cannot register
* Users within the age range can register normally
* Users without a profile are redirected to create one for age-restricted events
* Admins cannot manually register ineligible users via the admin form
* Bulk confirmation action skips age-ineligible registrations
* Open events (18-99 range) don't require profile creation
* Boundary ages (exactly min or max) are accepted

## Other Information
The fix addresses a regression where age verification was bypassed entirely in the registration flow. The three-layer enforcement (view, form, bulk action) ensures consistency across all registration paths.

https://claude.ai/code/session_01LQv2BGmp9NF2k2HH6otAkB

### CODE REVIEW

File: crush_lu/views_events.py [hunk 1]
```diff
@@ -783,6 +803,37 @@ def event_register(request, event_id):
                     )
                     return redirect("crush_lu:event_detail", event_id=event_id)
 
+                # Defense-in-depth: re-verify age under lock against the freshly
+                # locked event, in case event.min_age / max_age or the user's
+                # DOB changed concurrently.
+                if event_has_age_restriction:
```
[the remark this answers is not in the document]
Comment (EuphoriaLux) [reply crush_lu/views_events.py]: Good catch — fixed in 53e5153. The lock-time block now derives the restriction flag from `locked_event.min_age` / `locked_event.max_age` inside the atomic block, so a concurrent tightening between the pre-lock read and `select_for_update()` is still enforced.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crush_lu/admin/events.py [hunk 2]
```diff
@@ -30,9 +31,64 @@
...
+
+    def clean(self):
+        cleaned_data = super().clean()
+        event = cleaned_data.get("event")
+        user = cleaned_data.get("user")
+        status = cleaned_data.get("status")
+
+        # Cancelled registrations don't need to pass the age gate.
+        if not event or not user or status == "cancelled":
+            return cleaned_data
```
[the remark this answers is not in the document]
Comment (EuphoriaLux) [reply crush_lu/admin/events.py:64]: Good catch — fixed in 53e5153. `clean()` now falls back to `self.instance.event` / `.user` / `.status` before deciding to skip, so the inline admin (implicit event) and changelist `list_editable` updates (only `status` / `payment_confirmed` posted) still run the age gate. Added two regression tests covering both cases: `test_admin_inline_form_cannot_bypass_via_implicit_event` and `test_admin_changelist_inline_edit_cannot_bypass`.

---
_Generated by [Claude Code](https://claude.ai/code)_
